server/api/Users.js

```diff
@@ -9,64 +9,74 @@ const {
 } = require('./Utils');
 // add require by token middleware
 
-// GET /api/users/all
-// add requireAdminToken util
+// GET /api/users/all returns all users if given a valid Admin token
 router.get('/all', requireAdminToken, async (req, res, next) => {
   try {
-    res.send(await User.findAll());
+    res.send(await User.findAll({ attributes: { exclude: ['password'] } }));
   } catch (er) {
     next(er);
   }
 });
 
-// GET /api/users
-router.get('/', requireUserToken, async (req, res, next) => {
-  const { user } = req;
-  try {
-    console.log('----->GET/ api/users', user);
-    res.send(user);
-  } catch (er) {
-    next(er);
-  }
-});
-
-// GET /api/users/:id/cart
-// retrieves a cart's associated products from database
-router.get('/:id/cart', requireCartToken, async (req, res, next) => {
-  const { cart } = req;
-  console.log('requireCartToken', cart);
+// DELETE /api/users/:id deletes a user if given a valid Admin token
+router.delete('/:id', requireAdminToken, async (req, res, next) => {
   try {
-    res.send(cart);
+    const {
+      params: { id },
+    } = req;
+    const user = await User.findByPk(id);
+    await user.destroy();
+    res.sendStatus(204);
   } catch (er) {
     next(er);
   }
 });
 
-// PUT /api/users/:id
-router.put('/:id', async (req, res, next) => {
+// GET /api/users returns the current user details with a valid User token
+router.get('/', requireUserToken, async (req, res, next) => {
+  const { user } = req;
   try {
-    const user = await User.findByPk(req.params.id);
-    res.status(200).send(await user.update(req.body));
+    // console.log('----->GET/ api/users', user);
+    res.send(user);
   } catch (er) {
     next(er);
   }
 });
```

Comment:
We get the desired user from the `requireUserToken` utility, so we can just send that back when a user requests their info